src/OAuth2Adapter.php

```diff
@@ -28,20 +28,22 @@ class OAuth2Adapter implements AuthenticationInterface
     protected $resourceServer;
 
     /**
-     * @var ResponseInterface
+     * @var callable
      */
-    protected $responsePrototype;
+    protected $responseFactory;
 
     /**
      * Constructor
      *
      * @param ResourceServer $resourceServer
      * @param ResponseInterface $responsePrototype
      */
-    public function __construct(ResourceServer $resourceServer, ResponseInterface $responsePrototype)
+    public function __construct(ResourceServer $resourceServer, callable $responseFactory)
```

Comment:
Please remove the whole PHPDoc on `__construct`. It's not updated and we are not using them when there is no any additional information in it.